src/server/components/dashboard/listsItems/helpers.ts

```diff
@@ -20,14 +20,15 @@ export async function redirectIfUnauthorised(req: Request, res: ListItemRes, nex
 
     if (!userHasAccessToList) {
       const err = new HttpException(403, "403", "User is not authorised to access this list.");
-      return next(err);
+      next(err);
+      throw new Error(`User with id ${req.user?.id} is not authorised to access this list.`);
```

Comment:
This might cause a weird race condition and crash the app, you'll have to double check though. `next` might cause the response to end - but `throw new Error(`..`)` might be caught by another middleware which is also attempting a res.end. 

you can most commonly see this behaviour if you forget to return after doing a res.render or redirect

Reply:
Ah I see, okay I think of another way to do this.

Reply:
```
next(err)
return;
```
should do it.